core/src/identity.rs

```diff
@@ -0,0 +1,226 @@
+                    "proof_type":"Ed25519Signature2018",
+                    "public_key_multibase":"CapabilityInvocation"
+                }],
+                "context":["default"],
+                "id":"default"
+            })
+    }
+
+    fn get_did() -> String {
+        String::from("123456789")
+    }
+
+    #[rstest::rstest]
+    #[case::created_successfully(
+        Some(Ok(())),
+        get_did(),
+        get_json_input_mock(),
+        true
+    )]
+    #[case::created_error(
+        Some(Err(crate::error::ResolverError{
+            message: "testErr".to_string(), 
+            kind: crate::error::ErrorKind::NetworkFailure})),
+        get_did(),
+        get_json_input_mock(),
+        false
+    )]
+
+    fn test_create_identity(
+        #[case] mock_create_response: Option<Result<(), crate::error::ResolverError>>,
+        #[case] did_doc: String,
+        #[case] did_document_input_mock: serde_json::Value,
+        #[case] expect_ok: bool,
+    ) -> Result<(), String> {
+        let mut resolver_mock = MockDIDResolver::default();
+
+        resolver_mock
+            .expect_create()
+            .with(
+                mockall::predicate::function(|did_doc: &String| -> bool {
+                    did_doc.clone().len() > 0
+                }),
+                mockall::predicate::function(move |doc_input: &serde_json::Value| -> bool {
+                    let did_doc_input: DidDocument =
+                        serde_json::from_value(doc_input.clone()).unwrap();
+                    let did_doc_mock: DidDocument =
+                        serde_json::from_value(did_document_input_mock.clone()).unwrap();
+
+                    if did_doc_input.id != did_doc_mock.id {
+                        return false;
+                    }
+                    if did_doc_input.context != did_doc_mock.context {
+                        return false;
+                    }
+                    if did_doc_input.assertion_method[0].public_key_multibase
+                        != did_doc_mock.assertion_method[0].public_key_multibase
+                    {
+                        return false;
+                    }
+                    return true;
+                }),
+            )
+            .return_once(|_, _| (mock_create_response.unwrap()));
+
+        let iu = Identity::new(resolver_mock);
+        let kp = signature::keypair::Ed25519SSIKeyPair::new();
+        let gn = iu.generate(kp);
+        let identity = aw!(gn);
+
+        match identity {
+            Ok(_) => {
+                if expect_ok {
+                    Ok(())
+                } else {
+                    Err("Expected error".to_string())
+                }
+            }
+            Err(_) => {
+                if expect_ok {
+                    Err("Expected success".to_string())
+                } else {
+                    Ok(())
+                }
+            }
+        }
```

Comment:
You can possible reduce nesting by using one match arm and an assertion:
https://github.com/mkatychev/darkroom/blob/9eaaec3bc63583125b67e2b1d25530385be790f9/src/record.rs#L307-L320

[`Result::is_err`](https://doc.rust-lang.org/std/result/enum.Result.html#method.is_err) is also good to panic on here depending on your match arm.